### PR TITLE
Add monolog - Second try

### PR DESCRIPTION
**Pull request by nupplaphil**
Followup after #6354 

Todos (follow up PR?):
- admin GUI for logs will set the legacy loglevel
- what about legacy loglevels in "local.config.php" ? I guess we cannot solve this problem automatically

### CODE REVIEW

File: src/Core/Logger.php [hunk 1]
```diff
@@ -5,82 +5,335 @@
...
+	{
+		$debugging = Config::get('system', 'debugging');
+		$logfile = Config::get('system', 'logfile');
+		$loglevel = Config::get('system', 'loglevel');
+
+		if (!$debugging || !$logfile) {
+			return;
+		}
+
+		if (is_int($loglevel)) {
```
Comment (nupplaphil) [inline src/Core/Logger.php]: This check is now necessary for legacy logleve vs. PSR-3 loglevels

**Comment by nupplaphil** [comment]
btw. I clicked a little bit on my local instance .. should now work :-/

File: config/settings.config.php [hunk 2]
```diff
@@ -66,6 +70,16 @@
...
 
+		// logfile (String)
+		// The logfile for storing logs.
+		// Can be a full path or a relative path to the Friendica home directory
+		'logfile' => 'friendica.log',
+
+		// loglevel (String)
+		// The loglevel for all logs.
+		// Has to be one of these values: emergency, alert, critical, error, warning, notice, info, debug
+		'loglevel' => 'notice',
```
Comment (nupplaphil) [inline config/settings.config.php]: Default loglevel as fallback and as a doc if somebody want to change the value

**Comment by AndyHee** [comment]
@nupplaphil this is a cool new feature by the sounds of it. I'm looking forward to the new gui!

It seems to work now! But I had to change the permission of my logfile. My wrong file permissions initially caused a fatal error.

What permissions are recommended for the logfile? I have a dedicated user for this server and only `www-data` for smarty.

```
2019/01/04 23:04:11 [error] 1133#1133: *25021 FastCGI sent in stderr: "PHP message: PHP Warning:  session_start(): Failed to read session data: user (path: /var/lib/php/sessions) in /var/www/friendica/src/App.php on line 1561PHP message: PHP Fatal error:  Uncaught UnexpectedValueException: The stream or file "/var/www/friendica/logs/friendica.log" could not be opened: failed to open stream: Permission denied in /var/www/friendica/vendor/monolog/monolog/src/Monolog/Handler/StreamHandler.php:107
Stack trace:
#0 /var/www/friendica/vendor/monolog/monolog/src/Monolog/Handler/AbstractProcessingHandler.php(39): Monolog\Handler\StreamHandler->write(Array)
#1 /var/www/friendica/vendor/monolog/monolog/src/Monolog/Logger.php(344): Monolog\Handler\AbstractProcessingHandler->handle(Array)
#2 /var/www/friendica/vendor/monolog/monolog/src/Monolog/Logger.php(609): Monolog\Logger->addRecord(250, 'no data for ses...', Array)
#3 /var/www/friendica/src/Core/Logger.php(335): Monolog\Logger->log(250, 'no data for ses...')
#4 /var/www/friendica/src/Core/Session/DatabaseSessionHandler.php(38): Friendica\Core\Logger::log('no data for ses...', 'notice')
#5 [internal function]: Friendica\Core\Session\DatabaseSessionHandler->read('b0cvf26b4u0elm" while reading response header from upstream, client: 195.201.227.53, server: friendica.hubup.pro, request: "GET /objects/174591b4-565c-2f7e-c65a-5fa956709479 HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/php7.3-fpm.sock:", host: "friendica.hubup.pro"
```

**Comment by annando** [comment]
I haven't followed the whole PR, so some question afterwards: The ```auth_ejabberd.php``` is run by the ejabberd user - so it mustn't log into the same logfile than the other log processes.

**Comment by nupplaphil** [comment]
OK, I'll add it in the next PR. I've an idea about how wie can use different logger for different channels and different outputs :-)

**Pull request merged**